### DISCUSSION TITLE
Rule proposal: Directive values should always be passed using brackets

### DISCUSSION THREAD

**Discussion by luiscla27** [angular-eslint/angular-eslint#860] [New Rule Proposals]
**This rule proposal is related to an issue**, the new rule would be read like this:
_Every directive declared in the current NgModule, should always be used using brackets_

**Proposal**
Currently, you can do this:
```html
<button SomeCustomDirective="5" />
```
The expected would be this:
```html
<button [SomeCustomDirective]="5" />
```

**Description of the issue**

Without the brackets, the IDE doesn't have a way of knowing which variable type is expecting, so it's easy to sometimes pass a wrong value... especially when the directive is refactored (let's say the expected value is `Array<number>` due to refactorization changes, but before it was `number`) 

In my case, our team separated a bunch of angular components, each one to a single own module so we were able to call them lazy. And we didn't realize that there it was a lot of missing module imports from NgModule's until we ran the app. I think our case counts as refactorization.

**Feedback**
Please guide me if this is not the right place to ask for this new rule request. I posted in this repo because someone in the `eslint`  project advised me to report this feature request in the **Angular-specific ESLint plugin** ([see this issue](https://github.com/eslint/eslint/issues/14326))

(this whole rule request applies also for `@Input` values)

**Comment by JamesHenry** [comment]
I don't think this one is true any more, the types can be picked up fine: 
<img width="1206" alt="image" src="https://github.com/user-attachments/assets/777ad7a0-1034-4c9a-9b54-1313a9b706f3" />

I don't think based on the current information a general rule can be recommended. If the Angular Team provides explicit guidance around this we can revisit
